### PR TITLE
Adds Art Collecting

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request
Adds art collecting. There is now a verb in the OOC tab that lets you browse the art gallery. You can then buy any art piece from its owner for a price of 500 beecoins + 500 for every time it's changed hands. 1/4 of the sale price goes to the original artist, 1/4 goes to the person you bought it from, and 1/2 goes to the void. Additionally, if art is bought during the pregame, it sends a message in OOC describing the purchase. Each painting's initial owner is the artist.

Purchasing art is strictly nonprofitable, you will never make positive beecoins off of purchasing art. At best, you will make back slightly more than a quarter of the purchase price.

## Why It's Good For The Game
Good for artists because good art will change hands a lot, and send beecoins in their direction as an incentive for making good art.
Good for the players with a lot of beecoins because they can flex and build up a collection.
Good for the players with not many beecoins because it has zero (0) effect on the actual game.
Good for the game because it provides a large beecoin sink.

## Testing Photographs and Procedure
image
(Text is incorrect, and has since been corrected)

## Changelog
:cl: Ivniinvi
add: Art collecting! Use the Browse Art Gallery verb in the OOC tab to purchase art. The original artist gets 1/4 of the purchase price, and the current owner gets 1/4.
/:cl:

### CODE REVIEW

**Comment by Ivniinvi** [comment]
Huh, my friend logged into github on my computer briefly and it appears to have signed the commit as by him. I actually created the PR, will see about fixing that.

**Comment by Oscari-pog** [comment]
I am gonna draw NSFW furry art and sell it

**Comment by PowerfulBacon** [comment]
Did you just screenshot my NFT?

**Comment by Ivniinvi** [comment]
image
UI updated to give artists credit and explain the distribution of coins

**Comment by AnCopper** [comment]
Literally just NFT's.

**Comment by EvilDragonfiend** [comment]
oh my god lol

**Comment by PowerfulBacon** [comment]
Time to use my 200000 coins to buy every bit of art on the server

**Comment by williamcll** [comment]
Is this what they call NFTs

**Comment by Crossedfall** [comment]
The metacoin sink we needed 💰

File: code/controllers/subsystem/persistence.dm [hunk 1]
```diff
@@ -294,6 +295,42 @@ SUBSYSTEM_DEF(persistence)
 	for(var/obj/structure/sign/painting/P in painting_frames)
 		P.load_persistent()
 
+	json_file = file("data/pending_art_metacoin.json")
+	if(fexists(json_file))
+		pending_art_metacoin = json_decode(file2text(json_file))
+
+	if(paintings && paintings["library"] && paintings["library"][1] && !paintings["library"][1]["price"]) //this should only ever happen once - adding owner and price values to legacy paintings
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/persistence.dm]:
```suggestion
	if(paintings && length(paintings["library"]) && !paintings["library"][1]["price"]) //this should only ever happen once - adding owner and price values to legacy paintings
```

File: code/datums/art_browser.dm [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		to_chat(usr, "<span class='warning'>You don't have enough [CONFIG_GET(string/metacurrency_name)]s to buy this painting.</span>")
+		return
+	usr.client?.inc_metabalance(price * -1, TRUE, "Purchased [title].")
+	SSpersistence.add_art_payout(owner, round(price * 0.25))
+	SSpersistence.add_art_payout(author, round(price * 0.25))
+	if(SSticker.current_state != GAME_STATE_PLAYING) // for flexing purposes
+		to_chat(world, "<span class='nicegreen'>[usr.ckey] purchased [title] from [owner] for [price] [CONFIG_GET(string/metacurrency_name)]s!</span>")
+	SSpersistence.paintings["library"][params["selected"]]["owner"] = usr.ckey
+	SSpersistence.paintings["library"][params["selected"]]["price"] += 500
+	refresh_owned()
```
Comment (PowerfulBacon) [inline code/datums/art_browser.dm:59]: There should be a confirmation message to the user, and they should be informated that it only saves at the end of the round.

File: code/controllers/subsystem/persistence.dm [hunk 3]
```diff
@@ -294,6 +295,42 @@ SUBSYSTEM_DEF(persistence)
...
+		message_admins("Legacy paintings detected, adding owner and price values")
+		log_game("Legacy paintings detected, adding owner and price values")
+		for(var/entry in paintings["library"])
+			entry["owner"] = entry["author"]
+			entry["price"] = 500
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/persistence.dm:307]: Hm, this should probably have some safety against those 2 entries not existing when its actually used.

**Comment by Ivniinvi** [comment]
me when the checks randomly fail

**Pull request closed without merging**